ios/pushprovisioning/AddToWalletButtonView.swift

```diff
@@ -0,0 +1,148 @@
+                    ErrorType.Failed,
+                    "Missing parameters. `cardHolderName` must be supplied in the props to <AddToWalletButton />"
+                ) as? [AnyHashable : Any]
+            )
+            return
+        }
+        
+        if (cardHolderName.isEmpty) {
+            onCompleteAction!(
+                Errors.createError(
+                    ErrorType.Failed,
+                    "`cardHolderName` is required, but the passed string was empty"
+                ) as? [AnyHashable : Any]
+            )
+            return
+        }
+
+        let config = STPPushProvisioningContext.requestConfiguration(
+            withName: cardHolderName,
+            description: cardDescription as String?,
+            last4: cardLastFour as String?,
+            brand: Mappers.mapToCardBrand(cardBrand as String?)
+        )
+        
+        // We can use STPFakeAddPaymentPassViewController ONLY IN TEST MODE. If STPFakeAddPaymentPassViewController is
+        // used with a live mode card, the flow will fail and show a 'Signing certificate was invalid' error.
+        let controller = {
+            return self.testEnv ? STPFakeAddPaymentPassViewController(requestConfiguration: config, delegate: self) : PKAddPaymentPassViewController(requestConfiguration: config, delegate: self)
+        }()
+        
+        let vc = findViewControllerPresenter(from: UIApplication.shared.delegate?.window??.rootViewController ?? UIViewController())
+        vc.present(controller!, animated: true, completion: nil)
+    }
+}
+
+
+extension AddToWalletButtonView: PKAddPaymentPassViewControllerDelegate {
+    func addPaymentPassViewController(_ controller: PKAddPaymentPassViewController, generateRequestWithCertificateChain certificates: [Data], nonce: Data, nonceSignature: Data, completionHandler handler: @escaping (PKAddPaymentPassRequest) -> Void) {
+        self.pushProvisioningContext = STPPushProvisioningContext(keyProvider: self)
+
+        self.pushProvisioningContext?.addPaymentPassViewController(controller, generateRequestWithCertificateChain: certificates, nonce: nonce, nonceSignature: nonceSignature, completionHandler: handler);
+    }
+
+    func addPaymentPassViewController(_ controller: PKAddPaymentPassViewController, didFinishAdding pass: PKPaymentPass?, error: Error?) {
+        if let error = error as NSError? {
+            onCompleteAction!(
+                Errors.createError(
+                    error.code == PKAddPaymentPassError.userCancelled.rawValue ? ErrorType.Canceled : ErrorType.Failed,
+                    error as NSError?
+                ) as? [AnyHashable : Any]
+            )
+        } else {
+            onCompleteAction!([
+                "error": NSNull(),
+            ] as [AnyHashable : Any])
+        }
+        let vc = findViewControllerPresenter(from: UIApplication.shared.delegate?.window??.rootViewController ?? UIViewController())
+        vc.dismiss(animated: true, completion: nil)
```

Comment:
Dismissing `controller` can be safer here.

```suggestion
        controller.dismiss(animated: true, completion: nil)
```